contributors/design-proposals/local-storage-pv.md

```diff
@@ -0,0 +1,632 @@
+        // The full path to the volume on the node
+        // For alpha, this path must be a directory
+        // Once block as a source is supported, then this path can point to a block device
+        Path string
+}
+
+type PersistentVolumeSource struct {
+    <snip>
+    // Local represents directly-attached storage with node affinity.
+    // +optional
+    Local *LocalVolumeSource
+}
+```
+
+The relationship between a Local volume and its node will be expressed using
+PersistentVolume node affinity, described in the following section.
+
+Users request Local volumes using PersistentVolumeClaims in the same manner as any
+other volume type. The PVC will bind to a matching PV with the appropriate capacity,
+AccessMode, and StorageClassName.  Then the user specifies that PVC in their
+Pod spec.  There are no special annotations or fields that need to be set in the Pod
+or PVC to distinguish between local and remote storage.  It is abstracted by the
+StorageClass.
+
+#### PersistentVolume Node Affinity
+
+PersistentVolume node affinity is a new concept and is similar to Pod node affinity,
+except instead of specifying which nodes a Pod has to be scheduled to, it specifies which nodes
+a PersistentVolume can be attached and mounted to, influencing scheduling of Pods that
+use local volumes.
+
+For a Pod that uses a PV with node affinity, a new scheduler predicate
+will evaluate that node affinity against the node's labels.  For this initial phase, the
+PV node affinity is only considered by the scheduler for already-bound PVs.  It is not
+considered during the initial PVC/PV binding, which will be addressed in a future release.
+
+Only the `requiredDuringSchedulingIgnoredDuringExecution` field will be supported.
+
+##### API Changes
+
+For the initial alpha phase, node affinity is expressed as an optional
+annotation in the PersistentVolume object.
+
+```
+// AlphaStorageNodeAffinityAnnotation defines node affinity policies for a PersistentVolume.
+// Value is a string of the json representation of type NodeAffinity
+AlphaStorageNodeAffinityAnnotation = "volume.alpha.kubernetes.io/node-affinity"
+```
+
+#### Local volume initial configuration
+
+There are countless ways to configure local storage on a node, with different patterns to
+follow depending on application requirements and use cases.  Some use cases may require
+dedicated disks; others may only need small partitions and are ok with sharing disks.
+Instead of forcing a partitioning scheme on storage administrators, the Local volume
+is represented by a path, and lets the administrators partition their storage however they
+like, with a few minimum requirements:
+
+* The paths to the mount points are always consistent, even across reboots or when storage
+is added or removed.
+* The paths are backed by a filesystem (block devices or raw partitions are not supported for
+the first phase)
+* The storage is mounted according to the provisioner discovery scheme defined below
```

Comment:
Is this a minimal requirement?  My understanding is that admin can use local volume without provisioner so doesn't have to conform to provisioner discovery scheme.

Reply:
Yes, thanks for catching, will remove.